tools/codegen/src/generator/languages/rust_lazy_reader/generator.rs

```diff
@@ -0,0 +1,718 @@
+        Ok(())
+    }
+}
+
+impl LazyReaderGenerator for ast::DynVec {
+    fn gen_rust<W: io::Write>(&self, output: &mut W) -> io::Result<()> {
+        let (transformed_name, tc) = get_rust_type_category(self.item().typ());
+        generate_rust_common_array(
+            output,
+            self.name(),
+            transformed_name,
+            tc,
+            None,
+            None,
+            Some(self),
+        )?;
+
+        let verify_sub = {
+            let func = verify_typ(self.item().typ().as_ref(), quote!(self.get(i)?));
+            if func.is_empty() {
+                quote!()
+            } else {
+                quote!(for i in 0..self.len()? {
+                    #func
+                })
+            }
+        };
+        let val_compatible = if verify_sub.is_empty() {
+            quote!(_compatible)
+        } else {
+            quote!(compatible)
+        };
+
+        let name = ident_new(self.name());
+        let q = quote! {
+            impl #name {
+                pub fn verify(&self, #val_compatible: bool) -> Result<(), Error> {
+                    self.cursor.verify_dynvec()?;
+                    #verify_sub;
+                    Ok(())
+                }
+            }
+        };
+        writeln!(output, "{}", q)?;
+        Ok(())
+    }
+}
+
+impl LazyReaderGenerator for ast::Table {
+    fn gen_rust<W: io::Write>(&self, output: &mut W) -> io::Result<()> {
+        generate_rust_common_table(output, self.name(), self.fields(), None)?;
+        let field_count = self.fields().len();
+        let name = ident_new(self.name());
+
+        let verify_fields = self
+            .fields()
+            .iter()
+            .enumerate()
+            .map(|(_i, f)| verify_filed(f));
+
+        let q = quote! {
+            impl #name {
+                pub fn verify(&self, compatible: bool) -> Result<(), Error> {
+                    self.cursor.verify_table(#field_count, compatible)?;
+                    #( #verify_fields )*
+                    Ok(())
+                }
+            }
+        };
+        writeln!(output, "{}", q)?;
+        Ok(())
+    }
+}
+
+fn generate_rust_common_array<W: io::Write>(
+    output: &mut W,
+    plain_name: &str,
+    type_name: TokenStream,
+    tc: TypeCategory,
+    array: Option<&ast::Array>,
+    fixvec: Option<&ast::FixVec>,
+    _dynvec: Option<&ast::DynVec>,
+) -> io::Result<()> {
+    let name = ident_new(plain_name);
+    let q = quote! {
+        pub struct #name {
+            pub cursor: Cursor,
+        }
+
+        impl From<Cursor> for #name {
+            fn from(cursor: Cursor) -> Self {
+                Self { cursor }
+            }
+        }
+    };
+
+    writeln!(output, "{}", q)?;
+
+    // len
+    if let Some(arr) = array {
+        let item_count = Literal::usize_unsuffixed(arr.item_count());
+        let q = quote! {
+            impl #name {
+                pub fn len(&self) -> usize { #item_count }
```

Comment:
I recommend we also add a `total_size` method returning the total byte size of current molecule data structure.

Also, it might be good to follow conventions, so `item_count` also return the same data as `len`, see [here](https://docs.rs/ckb-types/latest/ckb_types/packed/struct.Byte32Vec.html) for an example.